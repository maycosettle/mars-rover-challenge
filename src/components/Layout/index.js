export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
