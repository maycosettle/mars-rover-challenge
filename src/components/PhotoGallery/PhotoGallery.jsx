import PropTypes from 'prop-types';

import useStyles from './styles';

const PhotoGallery = ({ photos }) => {
  const styles = useStyles();

  return (
    <ul role="list" className={styles.container}>
      {photos.map((photo) => (
        <li key={photo.src} className={styles.list}>
          <div className={styles.imageContainer}>
            <img src={photo.src} alt="mars" className={styles.image} />
          </div>
          <div className={styles.description}>
            <p className={styles.item}>Rover: {photo.rover.name}</p>
            <p className={styles.item}>Status: {photo.rover.status}</p>
            <p className={styles.item}>Camera: {photo.camera.name}</p>
            <p className={styles.item}>Sol: {photo.sol}</p>
            <p className={styles.item}>Earth_date: {photo.earth_date}</p>
          </div>
        </li>
      ))}
    </ul>
  );
};

PhotoGallery.propTypes = {
  photos: PropTypes.array,
};

PhotoGallery.defaultProps = {
  photos: [],
};

export default PhotoGallery;
