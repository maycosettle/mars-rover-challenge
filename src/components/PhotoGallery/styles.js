const useStyles = () => ({
  container: 'grid grid-cols-2 gap-x-4 gap-y-8 sm:grid-cols-3 sm:gap-x-6 lg:grid-cols-4 xl:gap-x-8',
  list: 'relative',
  imageContainer: 'group block w-full aspect-w-10 aspect-h-7 rounded-lg bg-gray-100 overflow-hidden',
  image: 'object-cover pointer-events-none group-hover:opacity-75',
  description: 'grid grid-cols-2',
  item: 'mt-2 mr-2 block text-sm font-medium text-gray-500 truncate pointer-events-none',
});

export default useStyles;
