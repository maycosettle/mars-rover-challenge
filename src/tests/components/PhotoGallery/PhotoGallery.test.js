import { shallow } from 'enzyme';
import { PhotoGallery } from 'components/PhotoGallery';

describe('tests on <PhotoGallery />', () => {
  const wrapper = shallow(<PhotoGallery photos={[]} />);

  test('Should render PhotoGallery correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
