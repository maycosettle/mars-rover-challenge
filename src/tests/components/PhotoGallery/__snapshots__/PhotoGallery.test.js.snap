// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <PhotoGallery /> Should render PhotoGallery correctly 1`] = `
<ul
  className="grid grid-cols-2 gap-x-4 gap-y-8 sm:grid-cols-3 sm:gap-x-6 lg:grid-cols-4 xl:gap-x-8"
  role="list"
/>
`;
